jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * (C) Copyright 2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.graph.AsUndirectedGraph;
+import org.jgrapht.traverse.GraphIterator;
+import org.jgrapht.traverse.LexBreadthFirstIterator;
+import org.jgrapht.traverse.MaximumCardinalityIterator;
+
+import java.util.*;
+
+/**
+ * Allows testing chordality of a graph. The inspected {@code graph} is specified at construction time
+ * and cannot be modified. Currently chordality of a graph is tested via {@link MaximumCardinalityIterator}
+ * by default. When {@link IterationOrder#LEX_BFS} is specified as a second constructor parameter, this
+ * {@code ChordalityInspector} uses {@link LexBreadthFirstIterator} to compute perfect elimination order.
+ * <p>
+ * A <a href="https://en.wikipedia.org/wiki/Chordal_graph">chordal graph</a> is one in which all cycles of
+ * four or more vertices have a chord, which is an edge that is not part of the cycle but connects two vertices
+ * of the cycle.
+ * <p>
+ * A graph is chordal iff its the vertices can be arranged into a perfect elimination order. More than one perfect
+ * elimination order may exist for a given graph. Either maximum cardinality search or lexicographical breadth-first
+ * search can be used to produce such an order.
+ * <p>
+ * Both lexicographical BFS and maximum cardinality search run in $\mathcal{O}(|V| + |E|)$. Checking whether given order
+ * is the perfect elimination order via {@link ChordalityInspector#isPerfectEliminationOrder(List)} takes
+ * $\mathcal{O}(|V| + |E|)$ as well. So, overall time complexity of the method
+ * {@link ChordalityInspector#isChordal()} is $\mathcal{O}(|V| + |E|)$.
```

Comment:
Would it make sense to add a line of text stating why you would use BFS instead of LEX_BFS (or vice versa)?

Reply:
Actually, it was added (the note was "the maximum cardinality search is used as a default iterator because it performs better in practice" or something similar). This statement is based only on my personal tests (random instances and complete graphs were used). MCS performs 1.5 times better than LexBFS on average. As it is not the official result and the algorithms weren't tested on all types of graphs, @d-michail suggested removing this note.

It seems to me, the note could sound something like this "Both algorithms have linear time complexity. The default option for ChordalityInspector is MaximumCardinalityIterator. This can be changed with an appropriate constructor parameter".